moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/trajectory_monitor.h

```diff
@@ -54,9 +54,28 @@ MOVEIT_CLASS_FORWARD(TrajectoryMonitor);  // Defines TrajectoryMonitorPtr, Const
 class TrajectoryMonitor
 {
 public:
+  /**
+   * @brief      This class contains the rcl interfaces for easier testing
+   */
+  class MiddlewareHandle
+  {
+  public:
+    /**
+     * @brief      Destroys the object.
+     */
+    virtual ~MiddlewareHandle() = default;
+
+    /**
+     * @brief      Add sleep using rate
+     */
+    virtual void sleep() = 0;
+  };
+
   /** @brief Constructor.
    */
   TrajectoryMonitor(const CurrentStateMonitorConstPtr& state_monitor, double sampling_frequency = 0.0);
```

Comment:
I could be convinced otherwise, but I would think that while we're here, we would use proper doxygen comments, especially on the new ctor.